app/controllers/api/tags_controller.rb

```diff
@@ -0,0 +1,87 @@
+  # Uses Rails' RESTful routes (check 'rake routes' for the configured routes)
+  class TagsController < MainApiController
+    DEFAULT_FIELDS = [:id, :name, :description, :creator, :use].freeze
+
+    def index
+      @assignment = Assignment.find_by(id: params[:assignment_id])
+
+      respond_to do |format|
+        parent = @assignment || current_course
+        tags = parent.tags.includes(:role, :groupings).order(:name)
+
+        tag_info = tags.map do |tag|
+          {
+            id: tag.id,
+            name: tag.name,
+            description: tag.description,
+            creator: tag.role.display_name,
+            use: tag.groupings.size
+          }
+        end
+        format.xml do
+          render xml: tag_info
+        end
+        format.json do
+          render json: tag_info
+        end
+      end
+    end
+
+    # Creates a new instance of the tag.
+    def create
+      params.permit(:name, :description)
+      begin
+        assignment = current_course.assignments.find_by(id: params[:assignment_id])
+        new_tag = Tag.new(course: current_course, role: current_role, assessment: assignment, **self.tag_params)
+        if new_tag.save && params[:grouping_id]
+          grouping = assignment.groupings.find(params[:grouping_id])
+          grouping.tags << new_tag
+        end
+      rescue StandardError => e
+        render 'shared/http_status', locals: { code: '422', message: e.to_s }, status: :unprocessable_entity
+      else
+        render 'shared/http_status',
+               locals: { code: '201', message: HttpStatusHelper::ERROR_CODE['message']['201'] }, status: :created
+      end
+    end
+
+    def update
+      tag = Tag.find(params[:id])
+      if tag.nil?
+        render 'shared/http_status', locals: { code: '404', message: I18n.t('tags.not_found') }, status: :not_found
+      else
+        begin
+          name = params[:name]
+          desc = params[:description]
+
+          # not using tag_params so it can default to the previous name/description if nil
+          tag.update!(name: name || tag.name, description: desc || tag.description)
+        rescue StandardError => e
+          render 'shared/http_status', locals: { code: '422', message: e.to_s }, status: :unprocessable_entity
+        else
+          render 'shared/http_status',
+                 locals: { code: '200', message: HttpStatusHelper::ERROR_CODE['message']['200'] }, status: :ok
+        end
+      end
+    end
+
+    def destroy
+      tag = Tag.find_by(id: params[:id])
+      if tag.nil?
+        render 'shared/http_status', locals: { code: '404', message: I18n.t('tags.not_found') }, status: :not_found
+      else
+        tag.destroy!
+        render 'shared/http_status',
+               locals: { code: '200', message: HttpStatusHelper::ERROR_CODE['message']['200'] }, status: :ok
+      end
+    end
+
+    private
+
+    def tag_params
+      { name: params[:name] || '', description: params[:description] || '' }
```

Comment:
Don't specify default values. If the `:name` or `:description` fields aren't specified, they won't be included in the `create`/`update`, which is fine. (In the `create` case a validation error should be raised, which is also fine.)